modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/PersistentPageMemoryTableStorage.java

```diff
@@ -81,22 +80,36 @@ public void start() throws StorageException {
     protected PersistentPageMemoryPartitionStorage createPartitionStorage(int partId) throws StorageException {
         TableView tableView = tableCfg.value();
 
-        FilePageStore partitionFilePageStore = ensurePartitionFilePageStore(tableView, partId);
+        PartitionFilePageStore partitionFilePageStore = ensurePartitionFilePageStore(tableView, partId);
 
         CheckpointTimeoutLock checkpointTimeoutLock = dataRegion.checkpointManager().checkpointTimeoutLock();
 
         checkpointTimeoutLock.checkpointReadLock();
 
         try {
-            PartitionMeta partitionMeta = getOrCreatePartitionMeta(tableView, partId, partitionFilePageStore);
+            PersistentPageMemory persistentPageMemory = dataRegion.pageMemory();
 
-            TableFreeList tableFreeList = createTableFreeList(tableView, partId, partitionMeta);
+            int grpId = groupId(tableView);
+
+            PartitionMeta meta = partitionFilePageStore.meta();
+
+            if (meta.isCreated()) {
```

Comment:
Why do we need a "isCreated" flag? Makes absolutely no sense. You can compare stored ids with default values instead.

Reply:
Fix it